elastic/kibana
# [Infrastructure UI] Add URL state to Hosts View

Closes  [#141492](https://github.com/elastic/kibana/issues/141492)
## Summary
This PR adds `query`, `timeRange` and `filters` parameters to the URL state on the hosts view. URL parameters are updated after search filters are applied (after click on the "update" button.

## Testing
Different cases:
- Add new search criteria ( filter / time range / query ) and click on "update" - the URL should update
- Save a query and reload
- Load a saved query 
- Change an existing query

![image](https://user-images.githubusercontent.com/14139027/199047590-29e375fb-6909-424b-89c4-ef9193a77b10.png)

![image](https://user-images.githubusercontent.com/14139027/199046342-29fbfa76-0314-462b-b593-2c535112be09.png)

![image](https://user-images.githubusercontent.com/14139027/199046201-76ace0fa-8d17-4e1f-b36f-54a2419fb6af.png)

- Open the URL in a new browser tab/window - the filters should be added
